### PR TITLE
Container Services: Add API version 2019-04-30 for Azure RedHat OpenShift 

### PR DESCRIPTION
**Pull request by amanohar**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by amanohar** [comment]
@julienstroheker @ravbhatnagar PTAL.

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5699 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5699/commits/86b1ff963b1ddea633e8ae3f117e6c70a2c92c2e

**Comment by julienstroheker** [comment]
@AutorestCI rebuild azure-sdk-for-python

File: specification/containerservice/resource-manager/readme.go.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ go:
 
 ``` yaml $(go) && $(multiapi)
 batch:
+  - tag: package-2019-04-30
```
Comment (julienstroheker) [inline specification/containerservice/resource-manager/readme.go.md]: `tag: package-2019-04`

File: specification/containerservice/resource-manager/readme.go.md [hunk 2]
```diff
@@ -21,6 +22,16 @@ batch:
   - tag: package-2017-07
 ```
 
+### Tag: package-2019-04-30 and go
```
Comment (julienstroheker) [inline specification/containerservice/resource-manager/readme.go.md]: `package-2019-04`

File: specification/containerservice/resource-manager/readme.go.md [hunk 3]
```diff
@@ -21,6 +22,16 @@ batch:
...
+
+These settings apply only when `--package-2019-04-30 --go` is specified on the command line.
```
Comment (julienstroheker) [inline specification/containerservice/resource-manager/readme.go.md]: `--package-2019-04 --go``

File: specification/containerservice/resource-manager/readme.go.md [hunk 4]
```diff
@@ -21,6 +22,16 @@ batch:
...
+Please also specify `--go-sdk-folder=<path to the root directory of your azure-sdk-for-go clone>`.
+
+``` yaml $(tag)=='package-2019-04-30' && $(go)
```
Comment (julienstroheker) [inline specification/containerservice/resource-manager/readme.go.md]: `'package-2019-04'`

File: specification/containerservice/resource-manager/readme.md [hunk 5]
```diff
@@ -34,10 +34,20 @@ These are the global settings for the ContainerServices API.
 
 ``` yaml
 openapi-type: arm
-tag: package-2019-02
+tag: package-2019-04-30
```
Comment (julienstroheker) [inline specification/containerservice/resource-manager/readme.md]: `tag: package-2019-04`

File: specification/containerservice/resource-manager/readme.md [hunk 6]
```diff
@@ -34,10 +34,20 @@ These are the global settings for the ContainerServices API.
...
 ```
 
+### Tag: package-2019-04-30
```
Comment (julienstroheker) [inline specification/containerservice/resource-manager/readme.md]: ### Tag: package-2019-04

File: specification/containerservice/resource-manager/readme.md [hunk 7]
```diff
@@ -34,10 +34,20 @@ These are the global settings for the ContainerServices API.
...
 
+These settings apply only when `--tag=package-2019-04-30` is specified on the command line.
```
Comment (julienstroheker) [inline specification/containerservice/resource-manager/readme.md]: `--tag=package-2019-04`

File: specification/containerservice/resource-manager/readme.md [hunk 8]
```diff
@@ -34,10 +34,20 @@ These are the global settings for the ContainerServices API.
...
+
+```yaml $(tag) == 'package-2019-04-30'
```
Comment (julienstroheker) [inline specification/containerservice/resource-manager/readme.md]: `'package-2019-04'`

**Review by julienstroheker — COMMENTED** [review]
Package name should be `YYYY-MM`

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-04-30/examples/OpenShiftManagedClustersCreate_Update.json [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+              "name": "compute",
+              "role": "compute",
+              "count": 4,
+              "vmSize": "Standard_D4s_v3",
+              "osType": "Linux",
+              "subnetCidr": "10.0.0.0/24"
+            }
+          ],
+          "routerProfiles": [
+            {
```
Comment (julienstroheker) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-04-30/examples/OpenShiftManagedClustersCreate_Update.json]: are we missing info here ?

Comment (amanohar) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-04-30/examples/OpenShiftManagedClustersCreate_Update.json]: This is for created (httpStatus code `201`). Today we don't return anything during create here.

File: specification/containerservice/resource-manager/readme.md [hunk 10]
```diff
@@ -34,10 +34,20 @@ These are the global settings for the ContainerServices API.
 
 ``` yaml
 openapi-type: arm
-tag: package-2019-02
+tag: package-2019-04
 ```
 
+### Tag: package-2019-04
 
+These settings apply only when `--tag=package-2019-04` is specified on the command line.
```
Comment (julienstroheker) [inline specification/containerservice/resource-manager/readme.md]: You also need to add section for : 

```
### Tag: package-2019-04-only

These settings apply only when `--tag=package-2019-04-only` is specified on the command line.

``` yaml $(tag) == 'package-2019-04-only'
input-file:
- Microsoft.ContainerService/stable/2019-04-30/openShiftManagedClusters.json
```

around line 127

Comment (amanohar) [reply specification/containerservice/resource-manager/readme.md]: Thanks added.

**Comment by amanohar** [comment]
@AutorestCI rebuild azure-sdk-for-python

**Comment by amanohar** [comment]
@AutorestCI rebuild azure-sdk-for-python

**Comment by amanohar** [comment]
@AutorestCI rebuild azure-sdk-for-python

**Comment by amanohar** [comment]
@AutorestCI rebuild azure-sdk-for-python

**Comment by amanohar** [comment]
@veronicagg, Admin, how can I get `WaitForARMFeedback` label added to this PR?

**Comment by veronicagg** [comment]
@amanohar just did coincidentally :)

**Review by veronicagg — CHANGES_REQUESTED** [review]
This new API versions looks to have small changes, but since it's new version, I've added the WaitForARMFeedback label.
Python, Go and .Net are generating SDKs. Ruby, JS and Java may need to be updated to account for the new version.

**Comment by amanohar** [comment]
@veronicagg thanks for the response. At this time OSA (or AKS) does not support Ruby, JS and Java SDKs.

**Review by KrisBash — APPROVED** [review]
LGTM

**Comment by amanohar** [comment]
@veronicagg PTAL.
> @veronicagg thanks for the response. At this time OSA (and AKS) does not support Ruby, JS and Java SDKs.

**Review by veronicagg — APPROVED** [review]
Approving since changes look good. 
Regarding the languages you mentioned you're not supporting, by default we should be shipping all languages for the Azure Services in the repo, unless there's some exception for this one, you or SDK owner for the language missing may need to edit the corresponding readme file at the time of release.

**Comment by amanohar** [comment]
@veronicagg thanks! I will check with others in my team adding supporting for other languages.

My PR is blocked on Python SDK automation: https://github.com/Azure/azure-rest-api-specs/pull/5557/checks?check_run_id=96370247

Any idea what the issue could be and how to resolve it? Thanks.

**Comment by amanohar** [comment]
@AutorestCI rebuild azure-sdk-for-python

**Comment by amanohar** [comment]
Thanks @veronicagg!

**Pull request merged**